bibsearch

```diff
@@ -1,12 +1,12 @@
 #!/bin/sh
 
 CheckArg () {
-    doi_regex='[https://doi.org/]?10\.[0-9]{4}[0-9.]*/[^[:space:]]*'
+    doi_regex='(https://doi.org/)?10\.[0-9]{4}[0-9.]*/[^[:space:]]*'
     if echo "$1" | grep -E -q -o "$doi_regex"; then
 	doi="$1"
-    elif [ "$(xdg-mime query filetype "$1")" = "application/pdf" ]; then
+    elif [ "$(dd if="$1" count=4 bs=1 status=none)" = "%PDF" ]; then
 	doi="$(pdfinfo "$1" | grep -E -q -o "$doi_regex")"
-	[ -z "$doi" ] && info="$(pdfinfo "$1" | grep -io "Title:.*" | awk -F ':          ' '{print $2}')"
+	[ -z "$doi" ] && { info="$(pdfinfo "$1" | grep "^Title")"; info="${info#Title: *}"; }
```

Comment:
Why you add `{` and `}` in between everything?

Reply:
To group the two statements.  You could also write
```suggestion
	if [ -z "$doi" ]; then
		info="$(pdfinfo "$1" | grep "^Title")"
		info="${info#Title: *}"
	fi
```